impala/interface.py

```diff
@@ -176,45 +176,50 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             reraise(exc_type, exc_val, exc_tb)
 
 
+def _replace_simple_markers(marker, operation, string_parameters):
+    """
+    Replaces qmark and format markers in the given operation, from the
+    string_parameters list.
+
+    Raises ProgrammingError on wrong number of parameters or bindings.
+    """
+    param_count = len(string_parameters)
+    marker_count = operation.count(marker)
+    if marker_count == 0:
+        return operation, False
+    if param_count != marker_count:
+        raise ProgrammingError("Incorrect number of bindings supplied. "
+                               "The current statement uses %d, and "
+                               "there are %d supplied." %
+                               (marker_count, param_count))
+    operation = operation.replace(marker, '{}')
```

Comment:
Ah, yes, you are correct.  I was checking with 2.7 and 3.5 branches, but didn't do with 2.6 branch...